endpoints/main.go

```diff
@@ -28,17 +33,27 @@ var totalFailedUptimeChecksCounterVec *prometheus.CounterVec
 // Record the metrics into the appropriate prometheus register under the label specified by organizationName
 // recordLongRunningMetrics specifies wether or not to record information contained in the capability statment
 func getHTTPRequestTiming(urlString string, organizationName string, recordLongRunningMetrics bool) {
-	var resp, responeTime, err = querier.GetResponseAndTiming(urlString)
+	ctx := context.Background()
+	// Closing context if HTTP request and response processing is not completed within 30 seconds.
+	// This includes dropping the request connection if there's no reply within 30 seconds.
+	ctx, cancelFunc := context.WithDeadline(ctx, time.Now().Add(30*time.Second))
+	// Cancel the context once we are done with this function so that context does not remain in memory (causing a leak)
+	// We need to call cancelFunc in this context because we are still reading from the response Body in this context
+	defer cancelFunc()
```

Comment:
comment on line 41 not necessary any more. comment on line 40 probably not necessary. consider removing.

Reply:
done.